homeassistant/components/snmp/device_tracker.py

```diff
@@ -109,16 +100,43 @@ def __init__(self, config):
                 community, mpModel=SNMP_VERSIONS[DEFAULT_VERSION]
             )
 
-        self._target = target
         self.request_args: RequestArgsType | None = None
         self.baseoid = baseoid
         self.last_results = []
         self.success_init = False
 
+    @classmethod
+    async def create(cls, config):
+        """Asynchronously test the target device bftore fully initializing the scanner."""
+        self = cls.__new__(cls)
+        host = config[CONF_HOST]
+
+        try:
+            # Try IPv4 first.
+            target = await UdpTransportTarget.create(
+                (host, DEFAULT_PORT), timeout=DEFAULT_TIMEOUT
+            )
+        except PySnmpError:
+            # Then try IPv6.
+            try:
+                target = Udp6TransportTarget(
+                    (host, DEFAULT_PORT), timeout=DEFAULT_TIMEOUT
+                )
+            except PySnmpError as err:
+                _LOGGER.error("Invalid SNMP host: %s", err)
+                return None
+        self._target = target  # pylint: disable=attribute-defined-outside-init
+
+        self.__init__(config)  # pylint: disable=unnecessary-dunder-call
+        return self
```

Comment:
Maybe something like this? In this case we can give up calling `RuntimeError` in the constructor.
```suggestion
        instance = cls(config)
        instance._target = target

        return instance
```

And we can define in the constructor `self._target` type as:
```
self._target: UdpTransportTarget | Udp6TransportTarget
```